File: 2785-sort-vowels-in-a-string/2785-sort-vowels-in-a-string.js
/**
 * @param {string} s
 * @return {string}
 */
var sortVowels = function(s) {

    let vowels = s.split("").map(c => c.charCodeAt(0)).filter(c => isVowel(String.fromCharCode(c)))
    
    vowels.sort((a,b) => a-b);
        
    let res = ""
    j=0
    for (const [i, c] of s.split("").entries() ) {
       
        if (isVowel(c)) {
           
            res +=String.fromCharCode(vowels[j])
            j++
        } else {
            res+=c
        }
    }
    
    return res
};


function isVowel(c) {
    return c == 'a' || c == 'e' || c == 'o'|| c == 'u'|| c == 'i' || 
           c == 'A' || c == 'E' || c == 'O'|| c == 'U'|| c == 'I';
}

File: 2785-sort-vowels-in-a-string/2785-sort-vowels-in-a-string.rs
use std::collections::VecDeque;

const CHARS: [char; 10] = ['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'];

impl Solution {

    pub fn sort_vowels(s: String) -> String {
        let mut vowels = s
            .chars()
            .filter(|a| CHARS.contains(a))
            .collect::<VecDeque<_>>();

        vowels.make_contiguous().sort();

        s.chars()
            .map(|c| {
                CHARS
                    .contains(&c)
                    .then(|| vowels.pop_front())
                    .unwrap_or(Some(c))
            })
            .filter_map(|f| f)
            .collect()
    }
}